modules/core/shared/src/main/scala/coursier/core/Resolution.scala

```diff
@@ -592,11 +592,12 @@ object Resolution {
   def forceScalaVersion(sv: String): Dependency => Dependency = {
 
     val sbv = sv.split('.').take(2).mkString(".")
+    val scala = if (sbv.startsWith("3")) "scala3" else "scala"
 
     val scalaModules = Set(
-      ModuleName("scala-library"),
```

Comment:
Maybe we could not force a version for `scala-reflect` and `scalap`, when Scala 3 is used, like
```scala
val scalaModules =
  if (sbv.startsWith("3."))
    Set(
      "scala3-library",
      "scala3-compiler"
    )
  else
    Set(
      "scala-library",
      "scala-compiler",
      "scala-reflect",
      "scalap"
    )
```